### ISSUE TITLE
`detect_format` misclassifies ng+Nygard combination as MADR4

### ISSUE THREAD

**Issue by toobuntu** [joshrotenberg/mdbook-lint#408]
Encountered during `adrs` Homebrew formula development. 

`detect_format()` in `crates/mdbook-lint-rulesets/src/adr/format.rs` is content-based only. If a file starts with `---`, then it is classified as `AdrFormat::Madr4` and anything else as `AdrFormat::Nygard`.

This then conflicts with the documented ng mode output (per [`book/src/compatibility.md`](https://github.com/joshrotenberg/adrs/blob/main/book/src/compatibility.md) in the adrs repository) of YAML frontmatter combined with Nygard section headings (`## Context`, `## Decision`, `## Consequences`).

A ng+Nygard file starts with `---`, so the linter misclassifies it as `AdrFormat::Madr4` and applies MADR rules. ADR004 then requires `## Context and Problem Statement` and ADR005 requires `## Decision Outcome`, which the Nygard template does not produce. The result is false positives that incorrectly block `adrs doctor`.

Consider distinguishing between: YAML frontmatter + MADR sections → `AdrFormat::Madr4`, YAML frontmatter + Nygard sections → `AdrFormat::NygardNg` (new variant) and no frontmatter + Nygard sections → `AdrFormat::Nygard`, and accepting `source_format` metadata from the caller
(e.g., `adrs-core`'s `lint_adr`) to bypass content-based heuristics when the format is already known from the repo config.

## Observed behavior

```
adrs --ng init docs/decisions
adrs new --no-edit "Test Decision"   # default format = Nygard in ng mode
adrs doctor
error: [ADR004] ADR is missing '## Context and Problem Statement' section
error: [ADR005] ADR is missing '## Decision Outcome' section
Found 2 error(s), 0 warning(s), 0 info(s)
echo $?
1
```

The same result occurs after `adrs --ng new --format=nygard --no-edit "Test"`.

## AI disclosure

`format.rs` was traced with AI-assisted source review.
